pkg/storage/tree/tree.go

```diff
@@ -181,6 +189,62 @@ func (t *Tree) Iterate(cb func(key []byte, val uint64)) {
 	}
 }
 
+func copiedByteSlice(oldSlice []byte) []byte {
+	newSlice := make([]byte, len(oldSlice))
+	copy(newSlice, oldSlice)
+	return newSlice
+}
+
+func keyWithPrefix(node *treeNode, prefix []byte) []byte {
+	key := append(prefix, semicolon)
+	key = append(key, node.Name...)
+	// TODO why is this slice copy necessary?
+	// when omitted, some node names become corrupted
+	return copiedByteSlice(key)
```

Comment:
This is the slice copy that side-steps that weird function name mutation bug. I still haven't tracked down how the mutation would happen. Ideally, I'll track it down before the PR is done.

Reply:
If I understand this code correctly this is a common go mistake we all make. It's due to how slices are working internally in go. [Here's a good article on it](https://go.dev/blog/slices-intro). The gist is that slices are actually just pointers to arrays with the actual data and so what tends to happen is you have two slices pointing to the actual same data and when you later change it in one place, the other slice gets affected as well.

Reply:
And yeah, copying the data just like you did is a good solution in most cases.

Reply:
Right, I get that slices are references to arrays, but I just don't see where I'm changing any of that data. I seem to only be doing appends, which shouldn't affect the original slices or the underlying array.